src/compiler/checker.ts

```diff
@@ -9779,17 +9779,36 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     if (p.flags & SymbolFlags.Accessor && useAccessors) {
                         const result: AccessorDeclaration[] = [];
                         if (p.flags & SymbolFlags.SetAccessor) {
+                            let setter: Declaration | undefined;
+
+                            if (p.declarations) {
+                                for (const d of p.declarations) {

```

Comment:
I think you can actually write this as a `forEach`, which will stop iterating and return the first truthy value the callback does.

Reply:
thanks for the tip :P it wasn't meant to be re-reviewed yet but you were fast as always :)

it seems that `symbolToParameterDeclaration` can't be reused in full after all but I'll re-investigate this tomorrow and ping you when I'm done with it